viewer/src/public/createRevealManager.ts

```diff
@@ -0,0 +1,45 @@
+/*!
+ * Copyright 2020 Cognite AS
+ */
+
+import { LocalModelIdentifier, CdfModelIdentifier, ModelDataClient } from '@/utilities/networking/types';
+import { CdfModelDataClient } from '@/utilities/networking/CdfModelDataClient';
+import { CogniteClient } from '@cognite/sdk';
+import { createCadManager } from '@/datamodels/cad/createCadManager';
+import { createPointCloudManager } from '@/datamodels/pointcloud/createPointCloudManager';
+import { RevealManager } from './RevealManager';
+import { LocalModelDataClient } from '@/utilities/networking/LocalModelDataClient';
+import { RevealOptions } from './types';
+import { initMetrics } from '@/utilities/metrics';
+import { omit } from 'lodash';
+
+export function createLocalRevealManager(revealOptions: RevealOptions = {}): RevealManager<LocalModelIdentifier> {
+  const modelDataClient = new LocalModelDataClient();
+  initMetrics(revealOptions.logMetrics !== false, 'local', {
+    moduleName: 'RevealManager',
+    methodName: 'constructor',
+    constructorOptions: omit(revealOptions, ['internal'])
+  });
+  return createRevealManager(modelDataClient, revealOptions);
+}
+export function createCdfRevealManager(
+  client: CogniteClient,
+  revealOptions: RevealOptions = {}
+): RevealManager<CdfModelIdentifier> {
+  initMetrics(revealOptions.logMetrics !== false, client.project, {
+    moduleName: 'RevealManager',
+    methodName: 'constructor',
+    constructorOptions: omit(revealOptions, ['internal'])
+  });
+  const modelDataClient = new CdfModelDataClient(client);
+  return createRevealManager(modelDataClient, revealOptions);
+}
+
+export function createRevealManager<T>(
+  client: ModelDataClient<T>,
+  revealOptions: RevealOptions = {}
+): RevealManager<T> {
+  const cadManager = createCadManager(client, revealOptions);
+  const pointCloudManager = createPointCloudManager(client);
+  return new RevealManager(cadManager, pointCloudManager);
```

Comment:
so if eventually constructor is called, why not to have initMetrics in that function or constructor directly?

Reply:
Yep, sorry, I noticed that there is no client in RevealManager itself.

Reply:
Since I didn't know the intention of mixpanel, I didn't want to change it's behavior :)

Reply:
The reason why it isn't in the constructur is that none of the info added to initmetric exists inside the current constructor.

Reply:
Well, I just would put these metrics into that createRevealManager to not copy it twice 🤷‍♂️ 

But with honest module/method names at least.

```
 initMetrics(revealOptions.logMetrics !== false, client.project || 'local', {
    moduleName: 'createRevealManager',
    methodName: 'createRevealManager',
    constructorOptions: omit(revealOptions, ['internal'])
  });
```